Account Onboard Utility/Accounts_Onboard_Utility.ps1

```diff
@@ -291,10 +291,12 @@ Function New-AccountObject
 		$_Account.secretManagement.automaticManagementEnabled = Convert-ToBool $AccountLine.enableAutoMgmt
 		if ($_Account.secretManagement.automaticManagementEnabled -eq $false)
 		{ $_Account.secretManagement.manualManagementReason = $AccountLine.manualMgmtReason }
-		$_Account.remoteMachinesAccess = "" | select "remoteMachines", "accessRestrictedToRemoteMachines"
-		$_Account.remoteMachinesAccess.remoteMachines = $AccountLine.remoteMachineAddresses
-		$_Account.remoteMachinesAccess.accessRestrictedToRemoteMachines = Convert-ToBool $AccountLine.restrictMachineAccessToList
-		
+		If($AccountLine.remotemachines -ne $Null -or "")
+		{
+			$_Account.remoteMachinesAccess = "" | select "remoteMachines", "accessRestrictedToRemoteMachines"
```

Comment:
From different issues, this line needs to stay in regardless if the AccountLine has remote machines or not
There is a known bug (I think in 10.4) that was solved in 11.x that trying to create an account without these properties in the account body would result an error

So my recommendation would be:
$_Account.remoteMachinesAccess = "" | select "remoteMachines", "accessRestrictedToRemoteMachines"
If(-not [String]::IsNullOrEmpty($AccountLine.remoteMachines))
{
                        $_Account.remoteMachinesAccess.remoteMachines = $AccountLine.remoteMachineAddresses
			$_Account.remoteMachinesAccess.accessRestrictedToRemoteMachines = Convert-ToBool $AccountLine.restrictMachineAccessToList
}

Reply:
@AssafMiron 

I am a little confused if all of what is in the script is required:

1) For the following:

$_Account.remoteMachinesAccess = "" | select "remoteMachines", "accessRestrictedToRemoteMachines"

Does this import the values in to the following variables:
$_Account.remoteMachinesAccess.remoteMachines
$_Account.remoteMachinesAccess.accessRestrictedToRemoteMachines 

If so, are the following two lines required/correct?
$_Account.remoteMachinesAccess.remoteMachines = $AccountLine.remoteMachineAddresses
$_Account.remoteMachinesAccess.accessRestrictedToRemoteMachines = Convert-ToBool $AccountLine.restrictMachineAccessToList

2) For the function 'Convert-ToBool':
-------
$retBool = $false
	
if($txt -match "^y$|^yes$") { $retBool = $true }
elseif ($txt -match "^n$|^no$") { $retBool = $false }
else { [bool]::TryParse($txt, [ref]$retBool) }
    
return $retBool
--------

Is the 'ElseIf' statement needed if '$retBool' is set to '$false' at the start of the function?

Reply:
Hey @NathTheDude 

1. The line:
$_Account.remoteMachinesAccess = "" | select "remoteMachines", "accessRestrictedToRemoteMachines"
Just creates the properties under remoteMachinesAccess, not the values
then the lines of:
$_Account.remoteMachinesAccess.remoteMachines = $AccountLine.remoteMachineAddresses
$_Account.remoteMachinesAccess.accessRestrictedToRemoteMachines = Convert-ToBool $AccountLine.restrictMachineAccessToList
Add the values from the CSV file (AccountLine is the current line in the CSV)

2. The logic behind the Convert-ToBool is that if the text is Yes/No then I can handle it one way (using the If and Else If)
and if it is something else (like True / False) I can try to parse it automatically
The try Parse will fail if it is not True / False and will result in False
So I do think that the Else if is needed to be sure I'm using a happy flow and not waiting for something to fail and count on that it will probably result how I intended :)

Makes sense?

Reply:
Hey@AssafMiron, thank you for the info - much appreciated.

Reply:
Should their be error handling in there in case the value does not equal: true, false, ^y, ^yes, ^n or ^N?

Reply:
@NathTheDude this is something that TryParse is already doing